mobile/lib/modules/album/services/album.service.dart

```diff
@@ -1,34 +1,107 @@
 import 'dart:async';
 
 import 'package:flutter/foundation.dart';
+import 'package:hive_flutter/hive_flutter.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/constants/hive_box.dart';
+import 'package:immich_mobile/modules/backup/models/hive_backup_albums.model.dart';
 import 'package:immich_mobile/shared/models/album.dart';
 import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/models/store.dart';
 import 'package:immich_mobile/shared/models/user.dart';
 import 'package:immich_mobile/shared/providers/api.provider.dart';
+import 'package:immich_mobile/shared/providers/db.provider.dart';
 import 'package:immich_mobile/shared/services/api.service.dart';
+import 'package:immich_mobile/shared/services/sync.service.dart';
+import 'package:immich_mobile/shared/services/user.service.dart';
+import 'package:isar/isar.dart';
 import 'package:openapi/api.dart';
+import 'package:photo_manager/photo_manager.dart';
 
 final albumServiceProvider = Provider(
   (ref) => AlbumService(
     ref.watch(apiServiceProvider),
+    ref.watch(userServiceProvider),
+    ref.watch(syncServiceProvider),
+    ref.watch(dbProvider),
   ),
 );
 
 class AlbumService {
   final ApiService _apiService;
+  final UserService _userService;
+  final SyncService _syncService;
+  final Isar _db;
+  Completer<bool> _localCompleter = Completer()..complete(false);
+  Completer<bool> _remoteCompleter = Completer()..complete(false);
 
-  AlbumService(this._apiService);
+  AlbumService(
+    this._apiService,
+    this._userService,
+    this._syncService,
+    this._db,
+  );
 
-  Future<List<Album>?> getAlbums({required bool isShared}) async {
+  Future<bool> refreshDeviceAlbums() async {
```

Comment:
Can we put a few words as the comment for this method and the `refreshRemoteAlbums` as the description of what each method do?

Reply:
sure! I've added some comments